### ISSUE TITLE
gitconfig.local

### ISSUE THREAD

**Issue by mrtnbroder** [notquitedante/dotfiles#4]
Hi there @danteissaias !

Awesome dotfiles! I really got inspired by them and I am therefore currently migrating my dotfiles to gnu-stow as well.

I do have some questions about your usage of gnu-stow and your local (public and private) files. Maybe you can help me understand a few things or explain to me how you use your dotfiles.

As far as I can tell, if we start fresh (meaning, a new macbook straight out of the box) would still require some prerequisites like `homebrew` and gnu `stow` already installed, right? So your dotfiles do not handle the inital bootstrapping. Is that intentional?

You published this file but without your name, email and signingkey:
https://github.com/danteissaias/dotfiles/blob/master/files/.gitconfig.local

How do you work with this file now, once you cloned your repository into `~/.dotfiles`? Do you just add in your name, email and signinkey and always have uncommited changes? How do you work on your dotfiles then?

Basically I was wondering how you keep private (local) files/changes on your local machine and do not accidentally push them to github. Hopefully you get my point and can help me out a bit!

On a little sidenote: currently `.gitconfig.local` would not be picked up by `.gitconfig` since the reference `~/.gitconfig.local` is not symlinked via stow. bug?

Thanks in advance, have a great weekend!

**Comment by danteissaias** [comment]
Hi, thanks for your interest in this project. I've been extremely busy at the moment and haven't had a chance to fix any bugs. It's intentional that you have to have `brew` and `stow` installed, although ideally I'd want to have a script that automatically clones etc, but I don't have the time right now to write it.  The whole deal with `.gitconfig.local` is that I was trying to make it work but it still doesn't work. In terms of keeping it private I just manually copy it to the home directory and change what is needed.

**Comment by notquitedante** [comment]
Hi,

I've finally done large refactor of my dotfiles. Currently the only dependency is that homebrew is already installed but I'll soon add a check in the `Makefile` to install homebrew if it's not already installed.

The whole deal with `.gitconfig.local` has been fixed. It's now automatically copied by the `Makefile` and all that's up to you is to put your name, email and signing key in there.
